### PR TITLE
Capture spew.Sprintf() with all our favorite config into a util func

### LINKED ISSUE #8976: Capture spew.Sprintf() with all our favorite config into a util func
util.PrettyPrintObject() or something.  Then find all the places that printf %#v on an object and use this instead.

### PR DESCRIPTION
**Pull request by haoruan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
A wrapper of spew.Sprintf, so to pretty print object with a util function

#### Which issue(s) this PR fixes:

Fixes #8976

#### Special notes for your reviewer:
With existing usage of spew.Sprintf, the wapper function print object with type info by %#v, and without type info by %+v

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Review by binacs — APPROVED** [review]
/ok-to-test

**Comment by haoruan** [comment]
Since some staged repositories are also using spew.Sprintf, so I decided to move it to kubernetes/utils, now is blocked by kubernetes/utils#226

File: pkg/util/print/print_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+func ptrint(i int) *int {
+	return &i
+}
+
+func ptrstr(s string) *string {
+	return &s
+}
+
+func TestPrettyPrintObject(t *testing.T) {
```
Comment (hgsigner) [inline pkg/util/print/print_test.go:31]: based on the previous suggestion, maybe we could write tests to assert `spew.Sprintf(format, a)` will be called with the right format.

**Comment by dims** [comment]
please reopen if you still need this WIP PR. Trying to declutter the open PR(s)

**Comment by haoruan** [comment]
reopen

**Pull request closed without merging**